src/taco-utils/promisesUtils.ts

```diff
@@ -0,0 +1,31 @@
+﻿/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../typings/lodash.d.ts" />
+
+"use strict";
+
+import _ = require("lodash");
+
+module TacoUtility {
+    export class PromisesUtils {
+        public static executeAfter<T>(valueOrPromise: T, afterCallback: { (): void }, failCallback?: { (reason: any): void }): T {
```

Comment:
The only difference is that a value won't be converted into a promise (this is used in telemetry.time) so:
var value = calculateSomethingSync();
can be converted to:
var value = telemetry.time(() => calculateSomethingSync()); // Here value is not a promise

And it also works for:
calculateSomethingAsync().then(doSomething);
which is converted to:
telemetry.time(() => calculateSomethingAsync()).then(doSomething); // Here the return value is a promise